packages/lpa-questionnaire-web-app/src/controllers/question-type/boolean.js

```diff
@@ -0,0 +1,66 @@
+const { VIEW } = require('../../lib/views');
+const getAppealSideBarDetails = require('../../lib/appeal-sidebar-details');
+const { createOrUpdateAppealReply } = require('../../lib/appeal-reply-api-wrapper');
+
+exports.BOOLEAN_VIEW = 'question-type/boolean';
+
+exports.getBooleanQuestion = (req, res) => {
+  const { id, heading, section, title } = res.locals.questionInfo;
+
+  const booleanInput = req.session.appealReply[id];
+
+  const values = {
+    booleanInput,
+  };
+
+  res.render(this.BOOLEAN_VIEW, {
+    appeal: getAppealSideBarDetails(req.session.appeal),
+    backLink: req.session.backLink || `/${req.params.id}/${VIEW.TASK_LIST}`,
+    values,
+    page: {
+      heading,
+      section,
+      title,
+    },
+  });
+};
+
+exports.postBooleanQuestion = async (req, res) => {
+  const { id, heading, section, title } = res.locals.questionInfo;
+  const {
+    body: { errors = {}, errorSummary = [], booleanInput = null },
+    session: { appealReply },
+  } = req;
+
+  const values = { booleanInput };
+  const page = {
+    heading,
+    section,
+    title,
+  };
+
+  try {
+    if (Object.keys(errors).length > 0) throw new Error('Validation failed');
```

Comment:
I can't see a test for this unhappy path.

Reply:
Not sure which tests you mean but Line 125 is the start of the test in packages/lpa-questionnaire-web-app/tests/unit/controllers/question-type/boolean.test.js for unit tests and AC02 in the e2e tests. They are there.